LaunchScripts/launchMaterials.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Start Docker Compose for PostgreSQL
+echo "Starting PostgreSQL for Materials Service"
+cd ..
+cd Materials
+docker-compose up -d
+
+# Wait for PostgreSQL to be ready
+sleep 3
```

Comment:
Sleep is not a reliable way. Try to use the script listed below. Also, you can reuse it for all services without duplicating this logic (at least to await required containers).

> [!IMPORTANT]
> You have to pass the container name as a parameter value

```sh
#!/bin/bash

# ANSI escape code for colors
CYAN="\033[36m"
RED="\033[31m"
GREEN="\033[32m"

RESET="\033[0m"

if [ $# -ne 1 ]; then
    echo "Usage: $0 <container-name>"
    exit 1
fi

container_name="$1"

echo "Starting Docker Container"
docker-compose up -d

timeout=$((SECONDS + 60))  # Set a timeout of 60 seconds
interval=1

while [ $SECONDS -lt $timeout ]; do
    # Check if PostgreSQL container is running
    if docker ps | grep -q "$container_name"; then
        echo -e "${GREEN} Container is running! ${RESET}"
        break
    else
        # Get current timestamp in [hh:mm:ss] format
        timestamp=$(date "+[%H:%M:%S]")
        echo -e "${CYAN}${timestamp} ${container_name} is not running. Retrying in $interval seconds... ${RESET}"
        sleep $interval
    fi
done

if [ $SECONDS -ge $timeout ]; then
    echo -e "${RED} ${container_name} did not start within a minute. Aborting. ${RESET}"
    exit 1
fi

exit 0
```

Usage:

```sh
./launch.sh postgres_1234
```

Reply:
Made improvements